### PR TITLE
Add Working with the API page

### LINKED ISSUE #1688: As a developer, I want reference documentation for the Query API endpoint
_Describe the issue here._

As a developer using the API, I want to be able to easily find examples showing the query API and a reference for the API. A google search returns unhelpful SEO results:

Screen Shot 2020-10-20 at 12 11 07 PM

The best of these is "Different ways to query influxdb": https://docs.influxdata.com/influxdb/v2.0/query-data/execute-queries/

From that page, there is a link to an API example - but no cross linking or reference to the actual API documentation. (Also, it only shows flux and not influxql).

Clicking the `query` tag below shows an unorganized group of pages.

From the full API documentation, https://docs.influxdata.com/influxdb/v2.0/api/, there is no preface or introductory material explaining anything about the API.  And clicking on "Query" (https://docs.influxdata.com/influxdb/v2.0/api/#operation/PatchDashboardsIDCellsIDView) from the ToC leads to a bunch of endpoints that are unlikely what the user wants (like PostQueryAst).

### PR DESCRIPTION
**Pull request by pierwill**
Closes #1688

Add a "Working with the API" page under Tools & integrations. Populate this page with some basic info and links to write and query task docs.

As we continue to build out developer-facing docs for the API, this page will serve as a landing page for a whole subsection under "Working with the API".

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: content/influxdb/v2.0/tools/api/_index.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+menu:
+  influxdb_2_0:
+    name: Working with the API
+    parent: Tools & integrations
+---
+
+InfluxDB offers a rich API that developers can use to create applications.
+The following pages offer general guides to the most commonly used API methods.
+
+For detailed documentation on the entire API, see [InfluxDBv2 API Documentation](/influxdb/v2.0/reference/api/#influxdb-v2-api-documentation)
```
Comment (kelseiv) [inline content/influxdb/v2.0/tools/api/_index.md:15]: How about line 15 to bottom?

File: content/influxdb/v2.0/tools/api/_index.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+---
```
Comment (kelseiv) [inline content/influxdb/v2.0/tools/api/_index.md:1]: duplicate OSS in cloud

File: content/influxdb/v2.0/tools/api/_index.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+title: Working with the InfluxDB API
+seotitle: Use the InfluxDB API
+description: Interact with InfluxDB using a rich API for writing and querying data and more.
+weight: 3
```
Comment (kelseiv) [inline content/influxdb/v2.0/tools/api/_index.md:5]: weight: 103

File: content/influxdb/v2.0/tools/api/_index.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+---
+title: Working with the InfluxDB API
+seotitle: Use the InfluxDB API
+description: Interact with InfluxDB using a rich API for writing and querying data and more.
```
Comment (kelseiv) [inline content/influxdb/v2.0/tools/api/_index.md:4]: InfluxDB 2.0

File: content/influxdb/v2.0/tools/api/_index.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+weight: 3
+menu:
+  influxdb_2_0:
+    name: Working with the API
+    parent: Tools & integrations
+---
+
+InfluxDB offers a rich API that developers can use to create applications.
```
Comment (kelseiv) [inline content/influxdb/v2.0/tools/api/_index.md:12]: that you

File: content/influxdb/v2.0/tools/api/_index.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+The following pages offer general guides to the most commonly used API methods.
```
Comment (kelseiv) [inline content/influxdb/v2.0/tools/api/_index.md:13]: Use the following API endpoints to write and query data:

File: content/influxdb/v2.0/tools/api/_index.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+For detailed documentation on the entire API, see [InfluxDBv2 API Documentation](/influxdb/v2.0/reference/api/#influxdb-v2-api-documentation)
+
+{{% note %}}
+If you are interacting with InfluxDB 1.x, see the [1.x compatibility API](/influxdb/v2.0/reference/api/influxdb-1x/).
+{{% /note %}}
+
+## Write API
+
+[Write data to InfluxDB](/influxdb/v2.0/write-data/developer-tools/api/) using an HTTP request to the InfluxDB API `/write` endpoint.
```
Comment (kelseiv) [inline content/influxdb/v2.0/tools/api/_index.md:23]: Maybe outside of scope of this PR, but should prob retitle this page we're linking to "Write data using API" being it doesn't cover query

**Review by kelseiv — APPROVED** [review]

File: content/influxdb/v2.0/tools/api/_index.md (hunk 1 above)
Comment (pierwill) [reply content/influxdb/v2.0/tools/api/_index.md:15]: At the moment, I think it would get lost under the query heading.

**Pull request merged**